File: app/src/main/java/com/example/my_plants/FriendWindow.java
package com.example.my_plants;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class FriendWindow extends AppCompatActivity {

    private static final String FRIEND_NAME = "total_count";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_home);
        showPlant();
    }

    protected void showPlant() {
        Bundle extras = getIntent().getExtras();
        String plant_name;

        if (extras == null) {
            plant_name = "MySon";
        } else {
            plant_name = extras.getString(FRIEND_NAME, "Dasha");
        }
        final TextView textView = (TextView)findViewById(R.id.plant_name_window);
        textView.setText(plant_name);

    }

}

File: app/src/main/java/com/example/my_plants/FriendWindow.kt
package com.example.my_plants

import android.os.Bundle
import android.view.View
import android.widget.TextView

import androidx.appcompat.app.AppCompatActivity

class FriendWindow : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_home)
        showPlant()
    }

    protected fun showPlant() {
        val extras = intent.extras
        val plant_name: String

        if (extras == null) {
            plant_name = "MySon"
        } else {
            plant_name = extras.getString(FRIEND_NAME, "Dasha")
        }
        val textView = findViewById<View>(R.id.plant_name_window) as TextView
        textView.text = plant_name

    }

    companion object {

        private val FRIEND_NAME = "total_count"
    }

}
